### ISSUE TITLE
加载DB_best.pth.tar失败

### ISSUE THREAD

**Issue by GraceKafuu** [BADBADBADBOY/pytorchOCR#65]
windows上跑了出问题，于是我想自己写inference

下面的 mobilenet_v3_small 是backbone/det_mobilev3.py里面的
------------------------------------------------------------------------------------------------------------
def mobilenet_v3_small(pretrained, **kwargs):
    model = MobileNetV3_Small()
    if pretrained:
        if torch.cuda.is_available():
            pretrained_dict = torch.load('./pre_model/mbv3_small.old.pth.tar')['state_dict']
        else:
            pretrained_dict = torch.load('./pre_model/mbv3_small.old.pth.tar', map_location='cpu')['state_dict']
        try:
            model.load_state_dict(pretrained_dict)
        except:
            state = model.state_dict()
            for key in state.keys():
                if 'module.' + key in pretrained_dict.keys():
                    state[key] = pretrained_dict['module.' + key]
            model.load_state_dict(state)
    return model
------------------------------------------------------------------------------------------------------------
model = mobilenet_v3_small(pretrained=True)
到这里时没问题的可以加载’./pre_model/mbv3_small.old.pth.tar‘ 成功，只测试了torch.randn(1, 3, 640, 640), 没测试图片，不知道图片有没有结果。

但是，
model = mobilenet_v3_small(pretrained=True)
model.load_state_dict(torch.load("checkpoint/DB_best.pth.tar")['state_dict'])
到这里出问题，我想问的是，是这么使用么？mbv3_small.old.pth.tar 和 DB_best.pth.tar 要一起使用么？还是单独使用一个就可以？

**Comment by LanWong1** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    出什么问题都没说
    
        
            
        
    
    
      
    
        
                    
            
                
                    
                            
                                
                            
                            
                                wangyi4422
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        
        
    
    
        签名由
        网易邮箱大师
        定制
    
 
    

    在2022年01月18日 ***@***.***> 写道： 




windows上跑了出问题，于是我想自己写inference

下面的 mobilenet_v3_small 是backbone/det_mobilev3.py里面的

def mobilenet_v3_small(pretrained, **kwargs):

model = MobileNetV3_Small()

if pretrained:

if torch.cuda.is_available():

pretrained_dict = torch.load('./pre_model/mbv3_small.old.pth.tar')['state_dict']

else:

pretrained_dict = torch.load('./pre_model/mbv3_small.old.pth.tar', map_location='cpu')['state_dict']

try:

model.load_state_dict(pretrained_dict)

except:

state = model.state_dict()

for key in state.keys():

if 'module.' + key in pretrained_dict.keys():

state[key] = pretrained_dict['module.' + key]

model.load_state_dict(state)

return model

model = mobilenet_v3_small(pretrained=True)

到这里时没问题的可以加载’./pre_model/mbv3_small.old.pth.tar‘ 成功，只测试了torch.randn(1, 3, 640, 640), 没测试图片，不知道图片有没有结果。

但是，

model = mobilenet_v3_small(pretrained=True)

model.load_state_dict(torch.load("checkpoint/DB_best.pth.tar")['state_dict'])

到这里出问题，我想问的是，是这么使用么？mbv3_small.old.pth.tar 和 DB_best.pth.tar 要一起使用么？还是单独使用一个就可以？



—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Comment by BADBADBADBOY** [comment]
> windows上跑了出问题，于是我想自己写inference
> 
> ## 下面的 mobilenet_v3_small 是backbone/det_mobilev3.py里面的
> ## def mobilenet_v3_small(pretrained, **kwargs):
> model = MobileNetV3_Small()
> if pretrained:
> if torch.cuda.is_available():
> pretrained_dict = torch.load('./pre_model/mbv3_small.old.pth.tar')['state_dict']
> else:
> pretrained_dict = torch.load('./pre_model/mbv3_small.old.pth.tar', map_location='cpu')['state_dict']
> try:
> model.load_state_dict(pretrained_dict)
> except:
> state = model.state_dict()
> for key in state.keys():
> if 'module.' + key in pretrained_dict.keys():
> state[key] = pretrained_dict['module.' + key]
> model.load_state_dict(state)
> return model
> model = mobilenet_v3_small(pretrained=True) 到这里时没问题的可以加载’./pre_model/mbv3_small.old.pth.tar‘ 成功，只测试了torch.randn(1, 3, 640, 640), 没测试图片，不知道图片有没有结果。
> 
> 但是， model = mobilenet_v3_small(pretrained=True) model.load_state_dict(torch.load("checkpoint/DB_best.pth.tar")['state_dict']) 到这里出问题，我想问的是，是这么使用么？mbv3_small.old.pth.tar 和 DB_best.pth.tar 要一起使用么？还是单独使用一个就可以？

你用mobile v3模型去加载dbnet的模型的参数当然不能加载了，mbv3_small.old.pth.tar 是dbnet 的mbv3 backbone的预训练参数，DB_best.pth.tar是我在icdar2015上训练好的dbnet模型的模型参数。

**Comment by BADBADBADBOY** [comment]
> windows上跑了出问题，于是我想自己写inference
> 
> ## 下面的 mobilenet_v3_small 是backbone/det_mobilev3.py里面的
> ## def mobilenet_v3_small(pretrained, **kwargs):
> model = MobileNetV3_Small()
> if pretrained:
> if torch.cuda.is_available():
> pretrained_dict = torch.load('./pre_model/mbv3_small.old.pth.tar')['state_dict']
> else:
> pretrained_dict = torch.load('./pre_model/mbv3_small.old.pth.tar', map_location='cpu')['state_dict']
> try:
> model.load_state_dict(pretrained_dict)
> except:
> state = model.state_dict()
> for key in state.keys():
> if 'module.' + key in pretrained_dict.keys():
> state[key] = pretrained_dict['module.' + key]
> model.load_state_dict(state)
> return model
> model = mobilenet_v3_small(pretrained=True) 到这里时没问题的可以加载’./pre_model/mbv3_small.old.pth.tar‘ 成功，只测试了torch.randn(1, 3, 640, 640), 没测试图片，不知道图片有没有结果。
> 
> 但是， model = mobilenet_v3_small(pretrained=True) model.load_state_dict(torch.load("checkpoint/DB_best.pth.tar")['state_dict']) 到这里出问题，我想问的是，是这么使用么？mbv3_small.old.pth.tar 和 DB_best.pth.tar 要一起使用么？还是单独使用一个就可以？

你用mobile v3模型去加载dbnet的模型的参数当然不能加载了，mbv3_small.old.pth.tar 是dbnet 的mbv3 backbone的预训练参数，DB_best.pth.tar是我在icdar2015上训练好的dbnet模型的模型参数。
